selinux/label_linux_test.go

```diff
@@ -0,0 +1,130 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package selinux_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/selinux"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type labelSuite struct {
+	path string
+}
+
+var _ = check.Suite(&labelSuite{})
+
+func (l *labelSuite) SetUpTest(c *check.C) {
+	l.path = filepath.Join(c.MkDir(), "foo")
+	ioutil.WriteFile(l.path, []byte("foo"), 0644)
+}
+
+func (l *labelSuite) TestVerifyHappyOk(c *check.C) {
+	cmd := testutil.MockCommand(c, "matchpathcon", "")
+	defer cmd.Restore()
+
+	ok, err := selinux.VerifyPathContext(l.path)
+	c.Assert(err, check.IsNil)
+	c.Assert(ok, check.Equals, true)
+	c.Assert(cmd.Calls(), check.DeepEquals, [][]string{
+		{"matchpathcon", "-V", l.path},
+	})
+}
+
+func (l *labelSuite) TestVerifyFailGibberish(c *check.C) {
+	cmd := testutil.MockCommand(c, "matchpathcon", "echo gibberish; exit 1 ")
+	defer cmd.Restore()
+
+	ok, err := selinux.VerifyPathContext(l.path)
+	c.Assert(err, check.ErrorMatches, "exit status 1")
+	c.Assert(ok, check.Equals, false)
+}
+
+func (l *labelSuite) TestVerifyFailStatus(c *check.C) {
+	cmd := testutil.MockCommand(c, "matchpathcon", "echo gibberish; exit 5 ")
+	defer cmd.Restore()
+
+	ok, err := selinux.VerifyPathContext(l.path)
+	c.Assert(err, check.ErrorMatches, "exit status 5")
+	c.Assert(ok, check.Equals, false)
+}
+
+func (l *labelSuite) TestVerifyFailNoPath(c *check.C) {
+	cmd := testutil.MockCommand(c, "matchpathcon", ``)
+	defer cmd.Restore()
+
+	ok, err := selinux.VerifyPathContext("does-not-exist")
+	c.Assert(err, check.ErrorMatches, ".* does-not-exist: no such file or directory")
+	c.Assert(ok, check.Equals, false)
+}
+
+func (l *labelSuite) TestVerifyFailNoTool(c *check.C) {
```

Comment:
this will fail on fedora like systems, no?  aren't we running unit tests there?

Reply:
Good point. I'll need to skip this iff the tool is already present. AFAIK we run unit tests only Ubuntu and Debian